### PR TITLE
cmd/recv.c: Migrate from deprecated gethostbyname() to getaddrinfo()

### PR DESCRIPTION
**Pull request by bernhardkaindl**
Commit message:

cmd/recv.c: Migrate from deprecated gethostbyname() to getaddrinfo()
    
The Fedora review-tool runs rpmlint on the binary packages and warned
that the uftrace tool was using gethostbyname. Switching to getaddrinfo
helps to get accepted, also for later uftrace distro packaging reviews:
    
POSIX.1-2008 removes the specifications of gethostbyname(), gethostbyaddr(),
and h_errno, recommending the use of getaddrinfo(3) and getnameinfo(3) instead:
    
They return a list of addrs or hostnames, are thread-safe and optionally
allow us to also use IPv6. This 1st step continues with IPv4 and only
still tries to connect to the primary, 1st address (for a small commit).

### CODE REVIEW

File: cmds/recv.c [hunk 1]
```diff
@@ -81,23 +81,45 @@ int setup_client_socket(struct uftrace_opts *opts)
...
+	// Just make sure the rest of your code is equally family-
+	// agnostic when dealing with the IP addresses associated
+	// with this connection. For instance, make sure any uses
+	// of sockaddr_in are changed to sockaddr_storage,
+	// and pay attention to its ai_ or ss_family field, etc.
+	//
+	// Next, we may loop over trying all addresses returned by getaddrinfo
+	// like in the example loops over the results list (accepted answer):
+	// IPv4 and IPv6 support is shown in answer 8 here:
+	// stackoverflow.com/questions/52727565/client-in-c-use-gethostbyname-or-getaddrinfo
```
Comment (namhyung) [inline cmds/recv.c:102]: Please use C style comments (`/* ... */`).

**Review by namhyung — COMMENTED** [review]
You forgot to add your Signed-off-by line at the end of the comment message.

File: cmds/recv.c (hunk 1 above)
Comment (bernhardkaindl) [reply cmds/recv.c]: Thanks, done!

File: cmds/recv.c [hunk 2]
```diff
@@ -81,23 +81,45 @@ int setup_client_socket(struct uftrace_opts *opts)
...
+	 *
+	 * After this, the code could be made IPv4/IPv6-agnostic: For this,
+	 * we'd change sockaddr_in to sockaddr_storage, check ai_family,
+	 * and use AF_UNSPEC in place of AF_INET which will allow getaddrinfo()
+	 * to return IPv4 and IPv6 addrs. (and the server would be updated too)
+	 *
+	 * A very nice example is shown in the accepted answer and answer 8 here:
+	 * stackoverflow.com/questions/52727565/client-in-c-use-gethostbyname-or-getaddrinfo
+	 */
+	memset(&hints, 0, sizeof hints);
```
Comment (namhyung) [inline cmds/recv.c:104]: For consistency, could you please use parentheses with `sizeof`?

**Review by namhyung — APPROVED** [review]
LGTM, but I have a nitpick.

**Pull request merged**